flask_common/crypto.py

```diff
@@ -38,51 +73,49 @@ def aes_generate_key():
 # Encrypt + sign using a random IV
 def aes_encrypt(key, data):
     assert len(key) == KEY_LENGTH, 'invalid key size'
-    iv = rng(AES_BLOCK_SIZE)
+    iv = rng(V0_IV_SIZE)
     return V0_MARKER + iv + aes_encrypt_iv(key, data, iv)
 
 
 # Verify + decrypt data encrypted with IV
-def aes_decrypt(key, data, extract_version=True):
+def aes_decrypt(key, data):
     assert len(key) == KEY_LENGTH, 'invalid key size'
-    extracted_version = None
-    if data[0] == V0_MARKER and extract_version:
-        extracted_version = V0_MARKER
-        data = data[1:]
-    iv = data[:AES_BLOCK_SIZE]
-    data = data[AES_BLOCK_SIZE:]
+
+    extracted_version = data[0]
+    data = data[1:]
+
+    if extracted_version == V0_MARKER:
+        iv = data[:V0_IV_SIZE]
+        data = data[V0_IV_SIZE:]
+    else:
+        iv = data[:IV_SIZE]
+        data = data[IV_SIZE:]
```

Comment:
Let's check for `V1_MARKER` and then if that isn't found raise an exception just so we know everything is working as expected.

Reply:
Done in https://github.com/closeio/flask-common/pull/75/commits/f1041c7573d188dedd19d5ed6c2391b0274e3d82.